src/hal/src/image.rs

```diff
@@ -208,6 +213,13 @@ pub enum Filter {
     Linear,
 }
 
+impl Default for Filter {
```

Comment:
why do we want a default filter here?

Reply:
Just to impl `Default` for `SamplerInfo`, which is something I wanted recently to make basic hello-world style setup just a little simpler.